main/src/library/Array.flix

```diff
@@ -1614,9 +1614,17 @@ namespace Array {
     /// of each element in `a` with `sep` inserted between each element.
     ///
     pub def join(sep: String, a: Array[a]): String & Impure with ToString[a] =
+        joinWith(x -> ToString.toString(x), sep, a)
+
+    ///
+    /// Returns the concatenation of the string representation
+    /// of each element in `a` according to `f` with `sep` inserted between each element.
+    ///
+    pub def joinWith(f: a -> String & ef, sep: String, a: Array[a]): String & Impure =
+        let lastSep = String.length(sep);
         let sb = StringBuilder.new();
-        foldLeft((_, x) -> StringBuilder.append!(sb, "${x}" + sep), (), a);
-        StringBuilder.toString(sb) |> String.dropRight(1)
+        foldLeft((_, x) -> StringBuilder.append!(sb, "${f(x)}" + sep), (), a);
```

Comment:
Its better (for performance) to call append! twice than to use +.

Reply:
(This may apply in other places too?)

Reply:
Roger